tests/unittests/test_request_timeout.py

```diff
@@ -0,0 +1,191 @@
+from tap_xero.client import XeroClient
+import unittest
+from unittest import mock
+from unittest.case import TestCase
+from requests.exceptions import Timeout, ConnectTimeout
+import datetime
+
+class TestBackoffError(unittest.TestCase):
+    '''
+    Test that backoff logic works properly.
+    '''
+    @mock.patch('tap_xero.client.requests.Request')
+    @mock.patch('tap_xero.client.requests.Session.send')
+    @mock.patch('tap_xero.client.requests.Session.post')
+    def test_backoff_check_platform_access_timeout_error(self, mock_post, mock_send, mock_request):
+        """
+        Check whether the request backoffs properly  for 60 seconds in case of Timeout error.
+        """
+        mock_send.side_effect = Timeout
+        mock_post.side_effect = Timeout
+        before_time = datetime.datetime.now()
+        with self.assertRaises(Timeout):
```

Comment:
I think these with assertRaises contexts should contain only the method that raises the exception, if the XeroClient init had a request in it and that timed out unexpectedly, this would never get to the check_platform_access step. But it would still pass. While I think it is performing functionally the way it's intended, it is difficult to see exactly what is being tested.

Reply:
Updated